mono/utils/mono-threads.c

```diff
@@ -1862,7 +1857,7 @@ mono_thread_info_wait_multiple_handle (MonoThreadHandle **thread_handles, gsize
 
 	res = mono_os_event_wait_multiple (thread_events, nhandles, waitall, timeout, alertable);
 	if (res >= MONO_OS_EVENT_WAIT_RET_SUCCESS_0 && res <= MONO_OS_EVENT_WAIT_RET_SUCCESS_0 + nhandles - 1)
-		return MONO_THREAD_INFO_WAIT_RET_SUCCESS_0 + (res - MONO_OS_EVENT_WAIT_RET_SUCCESS_0);
+		return (MonoThreadInfoWaitRet)(MONO_THREAD_INFO_WAIT_RET_SUCCESS_0 + (res - MONO_OS_EVENT_WAIT_RET_SUCCESS_0));
```

Comment:
I'd like a better record of all enum vs. int changes, so that if we really use C++ we can revisit them later more easily. But I guess I can grovel through the history, or do nothing, or make it into compile errors somehow, just to find them.